### PR TITLE
Change transaction acquisition mechanism

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
Suspend transactions in Resource

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
https://rchain.atlassian.net/browse/RCHAIN-3071

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by dzajkowski** [comment]
bors try

**Comment by KentShikama** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try-

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try-

**Comment by dzajkowski** [comment]
bors try

**Comment by ArturGajowy** [comment]
whelp, the merge build seems to have stuck: https://drone.rchain-dev.tk/rchain/rchain/1957/5

**Comment by dzajkowski** [comment]
bors try

**Comment by KentShikama** [comment]
If dev is broken with this deadlock anyway, might as well attempt a merge.

File: rspace/src/main/scala/coop/rchain/rspace/LMDBOps.scala [hunk 1]
```diff
@@ -44,6 +43,7 @@ trait LMDBOps[F[_]] extends CloseOps {
   private[this] val sizeGauge    = Kamon.gauge(MetricsSource + ".size").refine(gaugeTags)
   private[this] val entriesGauge = Kamon.gauge(MetricsSource + ".entries").refine(gaugeTags)
 
+  //fixme disabled for testing purposes
```
Comment (lukasz-golebiewski) [inline rspace/src/main/scala/coop/rchain/rspace/LMDBOps.scala]: I don't think this should go into `dev`

**Pull request merged**